### ISSUE TITLE
Raccoon training

### ISSUE THREAD

**Issue by kabrau** [experiencor/basic-yolo-keras#48]
Hi

First, congratulation for this implementation.

Im try train raccoon.
I had some problem, because I use Windows and Py35, but I solved everything.
I have GTX 1060 and 32GB ram

Now, my train does not improve, I tried full_yolo and mobilenet

Epoch 1/50
199/200 [============================>.] - ETA: 0s - loss: 0.2688Epoch 00000: val_loss improved from inf to 0.09791, saving model to my_MobileNet_raccoon2.h5
200/200 [==============================] - 77s - loss: 0.2678 - val_loss: 0.0979
Epoch 2/50
199/200 [============================>.] - ETA: 0s - loss: 1.0182Epoch 00001: val_loss did not improve
200/200 [==============================] - 72s - loss: 1.0150 - val_loss: 1.9232
Epoch 3/50
199/200 [============================>.] - ETA: 0s - loss: 0.5514Epoch 00002: val_loss did not improve
200/200 [==============================] - 74s - loss: 0.5503 - val_loss: 1.1664
Epoch 4/50
199/200 [============================>.] - ETA: 0s - loss: 0.4256Epoch 00003: val_loss did not improve
200/200 [==============================] - 74s - loss: 0.4245 - val_loss: 1.5531
Epoch 5/50
199/200 [============================>.] - ETA: 0s - loss: 0.3441Epoch 00004: val_loss did not improve
200/200 [==============================] - 75s - loss: 0.3440 - val_loss: 1.4772
Epoch 00004: early stopping

My config is

{
    "model" : {
        "architecture":         "MobileNet",
        "input_size":           416,
        "anchors":              [0.57273, 0.677385, 1.87446, 2.06253, 3.33843, 5.47434, 7.88282, 3.52778, 9.77052, 9.16828],
        "max_box_per_image":    10,        
        "labels":               ["raccoon"]
    },

    "train": {
        "train_image_folder":   "E:/Datasets/raccoon_dataset/train_image_folder/",
        "train_annot_folder":   "E:/Datasets/raccoon_dataset/train_annot_folder/",     
          
        "train_times":          10,
        "pretrained_weights":   "",
        "batch_size":           8,
        "learning_rate":        1e-4,
        "nb_epoch":             50,
        "warmup_batches":       250,

        "object_scale":         5.0 ,
        "no_object_scale":      1.0,
        "coord_scale":          1.0,
        "class_scale":          1.0,

        "saved_weights_name":   "my_MobileNet_raccoon2.h5",
        "debug":                false
    },

    "valid": {
        "valid_image_folder":   "E:/Datasets/raccoon_dataset/valid_image_folder/",
        "valid_annot_folder":   "E:/Datasets/raccoon_dataset/valid_annot_folder/",

        "valid_times":          1
    }
}

Can you help me ?
Tks

**Comment by experiencor** [comment]
I suggest you train in two steps:

1. Set warmup_batches to 250. Wait for the training to end.

2. Set warmup_batches to 0 and train again.

If it's still not working, you may play the 4 scales to increase recall (e.g. increase object_scale and coord_scale).

**Comment by kabrau** [comment]
Perfect !!
I retired EarlyStopping, and now improvef my trainning.

I have other question, do you now about metrics ? How I use in this case ?
I would like know accurace, confusion matrix.

Thank you again

**Comment by experiencor** [comment]
The standard metric of object detection is mean average precision (mAP). I still find time to implement the evaluation code. In the meantime, you can use the code from here https://github.com/marvis/pytorch-yolo2/blob/master/scripts/voc_eval.py.
